users/views.py

```diff
@@ -113,6 +115,24 @@ def delete(self, request):
         else:
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
+class UserFeedView(APIView):
+
+    def get(self, request):
+        user = IsLoggedIn(request)
+        if user is not None:
+            feed_posts_ids = list()
+            posts = Post.objects.all()
+            for post in posts:
+                if user in post.stream.followed_by.all():
+                    feed_posts_ids.append(post.id)
+                elif post.author in user.following.all():
+                    feed_posts_ids.append(post.id)
+            feed_posts = Post.objects.filter(id__in=feed_posts_ids)
+            serializer = PostSerializer(feed_posts, many=True)
+            return Response(serializer.data)
```

Comment:
Use a try block maybe in case of any exceptions ?

Reply:
objects.all() does not throw error in case of empty database, objects.get() does